pkg/retention/intervalCalender.go

```diff
@@ -0,0 +1,43 @@
+package retention
+
+import (
+	"fmt"
+)
+
+// intervalCalendar stores the time intervals of an interval-based retention strategy and validates the configuration
+type intervalCalendar struct {
+	name          StrategyId
+	timeIntervals []timeInterval
+}
+
+// InvalidCalendarConfigError contains the message and intervalName of an invalid retention strategy
+type invalidCalendarConfigError struct {
+	message      string
+	intervalName string
+}
+
+func (e *invalidCalendarConfigError) Error() string {
+	return fmt.Sprintf("%s: Please check the interval %s", e.message, e.intervalName)
+}
+
+func newIntervalCalendar(name StrategyId) *intervalCalendar {
+	return &intervalCalendar{name, nil}
+}
+
+func (ic intervalCalendar) addTimeIntervals(timeIntervals []timeInterval) intervalCalendar {
+	ic.timeIntervals = append(ic.timeIntervals, timeIntervals...)
+
+	return ic
+}
+
+func (ic intervalCalendar) validateConfig() error {
```

Comment:
🚫 **[checkstyle]** <unused> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>func `intervalCalendar.validateConfig` is unused